### PR TITLE
aws-janitor: use creation timestamp when available

### PR DESCRIPTION
**Pull request by ixdy**
Many resources don't have a readily accessible creation timestamp in their metadata, but at least some of them do, so we may as well make use of it when available.

/assign @detiber 
cc @chemikadze

### CODE REVIEW

File: aws-janitor/resources/instance.go [hunk 1]
```diff
@@ -54,8 +54,8 @@ func (Instances) MarkAndSweep(opts Options, set *Set) error {
 					Region:     opts.Region,
 					InstanceID: *inst.InstanceId,
 				}
-
-				if set.Mark(i) {
+				// Instances don't have a creation date, but launch time is better than nothing.
```
Comment (chemikadze) [inline aws-janitor/resources/instance.go:57]: Is that practically same thing? (I believe instances use LaunchXxx API call instead of CreateXxx)

**Review by chemikadze — APPROVED** [review]

File: aws-janitor/resources/instance.go (hunk 1 above)
Comment (ixdy) [reply aws-janitor/resources/instance.go:57]: I think for our purposes, they'll generally be the same thing, yes. My reference is [this SO thread](https://stackoverflow.com/questions/18916135/determining-amazon-ec2-instance-creation-date-time), which notes that instances can potentially be restarted (without being recreated), which would reset this timestamp, but I doubt most tests will be doing that.

**Review by richardcase — APPROVED** [review]
Seems ok to me, so:  

/lgtm

**Pull request merged**